### PR TITLE
Feature/1782 URLs in SMS notifications are shortened

### LINKED ISSUE #1782: URLs in SMS notifications are shortened
Part of epic #1017, dependent on #1781

AS a **volunteer/supervisor/admin**, 
I WANT to receive SMS notifications containing shortened links only,
SO THAT they are delivered to me in one message (not broken up due to length or blocked by my mobile carrier due to URL content).

### How it works now
Twilio is not set up yet, but once #1781 is complete, we will need to pick a link shortening service to integrate with Twilio. (Twilio used to offer their own link shortening, but that went away after Verizon blacklisted Twilio links in SMS.)

We don't have to use short.io, but they have <a href="https://blog.short.io/short-links-sms-marketing/">a blog post</a> with some good context on this.

### How it should work
Most mobile carriers block non branded shortened urls (i.e., bit.ly, rebrand.ly, etc), so we'll have to shop for a link shortening service that offers affordable branded options (there will likely be a cost associated with this). 

### Acceptance Criteria
- [x] create account with chosen link shortening service
- [x] integrate link shortening service with Twilio
- [x] enable branded links

### PR DESCRIPTION
**Pull request by xihai01**
### What github issue is this PR for, if any?
Resolves #1782 
short.io is chosen as the link shortening service and is integrated with Twilio.

### What changed, and why?
1. Added httparty gem -> used to simplify network requests in rails
2. Added twilio-ruby gem -> Twilio helper library for working with Twilio API easier
3. Created short_url_service.rb in app/services -> allows packaging short.io services into a class for easy use
4. Created twilio_service.rb in app/services -> allows packaging Twilio services into a class for easy use

### How will this affect user permissions?
- Volunteer permissions: N/A
- Supervisor permissions: N/A
- Admin permissions: N/A

### How is this tested? (please write tests!) 💖💪
Rspec tests are written in spec/services:
- A test is written to verify correct requests and response are being sent to/from short.io
- A test is written to verify an instantiated ShortUrlService object can return the short url link
- A test is written to verify the integration between Twilio and short.io -> Twilio can send a message with a short url

The request and response for these tests are stubbed using the webmock gem to avoid hitting the API's too much during tests.
Postman was used to verify the integrity and correct structure of the requests and responses being stubbed. (i.e correct headers, body params ...etc)

### Screenshots please :)
Postman request/responses for short.io and Twilio SMS API
Screenshot from 2022-05-11 22-53-07
Screenshot from 2022-05-08 16-01-09

### Feelings gif (optional)
Summer is almost here
corgi

### CODE REVIEW

File: app/services/short_url_service.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+require "json"
+
+class ShortUrlService
+  include HTTParty
+  base_uri "https://api.short.io/"
```
Comment (harsohailB) [inline app/services/short_url_service.rb:5]: nit: Should we place this in some sort of constants module? Perhaps something like [this](https://stackoverflow.com/questions/47153612/best-way-to-define-constants-in-ruby-ror-at-one-place)?

**Comment by harsohailB** [comment]
Really well done! Love the code cleanliness! 👌

File: app/services/short_url_service.rb (hunk 1 above)
Comment (xihai01) [reply app/services/short_url_service.rb:5]: I like that idea. I'll look into it

**Comment by xihai01** [comment]
created a file called constants.rb in config/initializers to hold API base urls and request header infos

**Review by 7riumph — APPROVED** [review]
Awesome, this is first-rate! The `twilio-ruby` and `twilio_service.rb` additions are also great steps towards completing this SMS project as a whole. Clean like @harsohailB said and well tested too 🔥

File: spec/services/short_url_service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+require "rails_helper"
+require "support/webmock_helper"
+
+RSpec.describe ShortUrlService do
+  describe "short.io API" do
+    before :each do
+      stubbed_requests
+      WebMock.disable_net_connect!
+      @original_url = "https://www.youtube.com/watch?v=dQw4w9WgXcQ"
```
Comment (7riumph) [inline spec/services/short_url_service_spec.rb:9]: 😂

File: app/services/short_url_service.rb (hunk 1 above)
Comment (xihai01) [reply app/services/short_url_service.rb]: resolved in 96eb0b0
let me know if this is what you were thinking of.

File: spec/services/short_url_service_spec.rb (hunk 2 above)
Comment (xihai01) [reply spec/services/short_url_service_spec.rb:9]: my first victim LOOL

File: app/services/short_url_service.rb (hunk 1 above)
Comment (harsohailB) [reply app/services/short_url_service.rb]: LGTM!

**Review by harsohailB — APPROVED** [review]
LGTM! Just have to fix the failing tests.

**Pull request merged**